.azure_pipelines/performance_check/run_performance_check.py

```diff
@@ -0,0 +1,245 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+# The Optimum optimization levels are:
+# O1: basic general optimizations.
+# O2: basic and extended general optimizations, transformers-specific fusions.
+# O3: same as O2 with GELU approximation.
+# O4: same as O3 with mixed precision (fp16, GPU-only, requires --device cuda).
+
+import argparse
+import ast
+import json
+import subprocess
+from pathlib import Path
+
+from optimum.exporters.onnx import main_export
+from optimum.onnxruntime import ORTOptimizer, ORTQuantizer
+from optimum.onnxruntime.configuration import AutoOptimizationConfig, AutoQuantizationConfig
+from tabulate import tabulate
+
+from olive.workflows import run as olive_run
+
+MODEL_NAME_MAP = {
+    "bert": "Intel/bert-base-uncased-mrpc",
+    "deberta": "microsoft/deberta-base-mnli",
+    "distilbert": "distilbert-base-uncased-finetuned-sst-2-english",
+    "roberta_large": "roberta-large-mnli",
+}
+
+MODEL_NAME_TO_CONFIG_MAP = {
+    "bert": {"model_name": "Intel/bert-base-uncased-mrpc", "task": "text-classification"},
+    "deberta": {"model_name": "microsoft/deberta-base-mnli", "task": "text-classification"},
+    "distilbert": {"model_name": "distilbert-base-uncased-finetuned-sst-2-english", "task": "text-classification"},
+    "roberta_large": {"model_name": "roberta-large-mnli", "task": "text-classification"},
+}
+
+ACC_METRIC = {
+    "name": "accuracy",
+    "type": "accuracy",
+    "backend": "huggingface_metrics",
+    "sub_types": [{"name": "accuracy", "priority": 1, "goal": {"type": "max-degradation", "value": 0.01}}],
+    "user_config": {
```

Comment:
For the metrics, it should be possible to add the data_configs directly to the metrics so that we can use the data container directly without user scripts? Similar to https://github.com/microsoft/Olive/blob/main/examples/open_llama/open_llama_sparsegpt_gpu.json#L59. 

That way, we don't need to maintain user scripts for each model. Just need the data configs which can be extracted from the input model in the olive workflow json.

Reply:
`huggingface_data_config_template(model_name, task, **kwargs).to_json()` will give us the data_config json to use in the template.

Reply:
Good call. I can add data config to model config itself.